### PR TITLE
Fix bug when responding to listings

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Sending a message to a listing author works about half the time. I
can't determine exactly what user behavior causes it to fail or succeed,
but after making this change it doesn't fail anymore. Looking at the
code, I'm not sure how it ever works...

Ater more manual testing, it seems like the failure has something to do with opening and closing the modals before responding. This change fixes the bug and will notify Honeybadger if things start acting out again.

## Related Tickets & Documents

Closes https://github.com/forem/forem/issues/12393

## QA Instructions, Screenshots, Recordings

Navigate to listings, attempt to respond to a listing. Try opening and closing multiple listing modals and responding, it shouldn't ever fail.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [x] No, and this is why: I am trying to get a Cypress E2E test working here, but have failed. :(
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: minor bugfix!

### CODE REVIEW

**Review by rhymes — APPROVED** [review]
Tested, it works

LGTM!

**Comment by nickytonline** [comment]
@jacobherrington, what issues are you having running Cypress? Happy to pair if you're blocked on writing E2E tests.

**Comment by jacobherrington** [comment]
@nickytonline Actually, I'm finding that a pretty huge amount of the time the existing Listing tests are failing (at least locally). I believe it's usually the second example, for some reason the Listing doesn't consistently render.

I'm OOO this week, but can definitely work together on it when I'm back.

**Comment by aitchiss** [comment]
> @nickytonline Actually, I'm finding that a pretty huge amount of the time the existing Listing tests are failing (at least locally). I believe it's usually the second example, for some reason the Listing doesn't consistently render.

Running it locally I'm seeing the same (wonder why we're not having this issue in CI?) - what's happening is that the listing created via `seeds_e2e` is being overwritten by the empty `/search/listings` results. I've pushed up a quick PR to fix that - #13395

**Comment by jacobherrington** [comment]
@aitchiss Thanks for the fix, that was my issue :smile: 

@nickytonline I added a Cypress test for this!

File: cypress/integration/listingFlows/viewListing.spec.js [hunk 1]
```diff
@@ -42,4 +42,22 @@ describe('View listing', () => {
     cy.get('@listingsModal').findByRole('button', { name: /Close/ }).click();
     cy.get('@listingTitle').should('have.focus');
   });
+
+  it('redirects when a user contacts via connect', () => {
```
Comment (nickytonline) [inline cypress/integration/listingFlows/viewListing.spec.js:46]: Documentation (blocking): Make it clearer in the test description that this is for a logged out  member trying to respond to a listing that gets redirected to the login. I don't know what the best wording is for this. Another option could be to put this it in a `describe` for a logged out member.

File: cypress/integration/listingFlows/viewListing.spec.js [hunk 2]
```diff
@@ -42,4 +42,22 @@ describe('View listing', () => {
...
+    cy.visit('/listings');
+    cy.findByRole('link', { name: 'Another listing' }).as('listingTitle');
+    cy.get('@listingTitle').click();
+    cy.findByTestId('listings-modal').as('listingsModal');
+    cy.findByTestId('listing-new-message').type('Hello there!');
+    cy.get('@listingsModal')
+      .findByRole('button', { name: /Send/ })
+      .as('sendButton');
+    cy.get('@sendButton').focus().click();
+    cy.url().should('not.include', '/listings');
```
Comment (nickytonline) [inline cypress/integration/listingFlows/viewListing.spec.js:61]: image

This test passes, but the check on the URL could be improved.

We could do for example 

`cy.url().should('include', '/connect/@admin_mcadmin');`

and also check that the page has the login form.

`cy.findByRole('main').findByTestId('login-form');`

http://localhost:3000/connect/@hegmannluigi

**Review by nickytonline — CHANGES_REQUESTED** [review]
This works locally for me. 😎

Just a couple change requests. Looking forward to seeing this in production!

File: cypress/integration/listingFlows/viewListing.spec.js (hunk 2 above)
Comment (jacobherrington) [reply cypress/integration/listingFlows/viewListing.spec.js]: Good idea, I wasn't quite sure on the right way to do this, but I like your suggestion!

**Review by Zhao-Andy — APPROVED** [review]
LGTM!

**Review by nickytonline — APPROVED** [review]

**Pull request merged**